lua/leetcode-ui/renderer/menu.lua

```diff
@@ -109,6 +109,20 @@ function Menu:apply_options()
     })
 end
 
+function Menu:_unmount()
+    for _, tabp in ipairs(require("leetcode.utils").question_tabs()) do
+        tabp.question:_unmount()
+    end
+
+    pcall(vim.api.nvim_buf_delete, self.bufnr, { force = true })
```

Comment:
(oops didn't mean to start a review)

Any reason you pcall here instead of just checking if the win/buf are valid?

Reply:
It might work with `buf_delete`, but it will still crash when trying to delete last window of neovim. If you have any better ideas feel free to fix this

Reply:
But in theory when using `Leet exit` we shouldn't be deleting the last window, because we create a new tab page. I just left it like this because it used to crash when new tabpage wasn't created

Reply:
You may not need to force close the window at all - If you look at the behavior of `nvim_buf_delete`, it closes the window containing the buffer unless it is the last window. If it is the last window, it switches to a different buffer.

I think you should only check if the buf is valid, then delete it. The window should take care of itself based on that.

(this `buf_delete` behavior is also why there are several plugins that provide custom bufdelete behavior)